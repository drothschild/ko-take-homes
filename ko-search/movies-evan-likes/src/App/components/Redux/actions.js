export const SET_SEARCH_FILTER = 'SET_SEARCH_FILTER'
export const SET_DECADE_FILTER = 'SET_DECADE_FILTER'
export const ADD_OMDB_DATA = 'ADD_OMDB_DATA'
export const EXPAND_LINE = 'EXPAND_LINE'
