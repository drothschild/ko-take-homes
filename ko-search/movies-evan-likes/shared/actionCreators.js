import axios from 'axios'
import { SET_SEARCH_FILTER, ADD_API_DATA, SET_DECADE_FILTER } from './actions'

export function setSearchFilter(searchFilter) {
  return { type: SET_SEARCH_FILTER, payload: searchFilter }
}

export function setDecadeFilter(decadeFilter) {
  return { type: SET_DECADE_FILTER, payload: decadeFilter }
}

export function addAPIData(apiData) {
  return { type: ADD_API_DATA, payload: apiData }
}

export function getOMDBDetails(title, year, id) {
  let dataWithID
  return dispatch => {
    axios
      .get(
        `http://www.omdbapi.com/?apikey=36ff4bcc&t=${encodeURIComponent(
          title
        )}&y=${year}&type=movie`
      )
      .then(response => {
        console.log(response.data)
        dataWithID = Object.assign({}, response.data)
        dataWithID.id = id
        dispatch(addAPIData(dataWithID))
      })
      .catch(error => {
        console.error('axios error', error)
      })
  }
}
