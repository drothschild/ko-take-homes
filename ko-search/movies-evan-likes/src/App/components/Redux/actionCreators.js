import axios from 'axios'
import { SET_SEARCH_FILTER, ADD_OMDB_DATA, SET_DECADE_FILTER, EXPAND_LINE } from './actions'

export function setSearchFilter(searchFilter) {
  return { type: SET_SEARCH_FILTER, payload: searchFilter }
}

export function setDecadeFilter(decadeFilter) {
  return { type: SET_DECADE_FILTER, payload: decadeFilter }
}

function addOMDBData(omdbData) {
  return { type: ADD_OMDB_DATA, payload: omdbData }
}

export function expandLine(expanded) {
  return { type: EXPAND_LINE, payload: expanded }
}

export function getOMDBDetails(title, year, id) {
  const dataWithID = {}
  if (localStorage.getItem([`${id}-imdb`]) && localStorage.getItem([`${id}-rating`])) {
    dataWithID.id = id
    dataWithID.imdbID = localStorage[`${id}-imdb`]
    dataWithID.rating = localStorage[`${id}-rating`]
    return addOMDBData(dataWithID)
  }
  return dispatch => {
    axios
      .get(
        `http://www.omdbapi.com/?apikey=36ff4bcc&t=
        ${encodeURIComponent(title)}&y=${year}&type=movie`
      )
      .then(response => {
        if (response.data.Ratings.filter(rating => rating.Source === 'Rotten Tomatoes')[0]) {
          dataWithID.rating = response.data.Ratings.filter(
            rating => rating.Source === 'Rotten Tomatoes'
          )[0].Value
        } else {
          dataWithID.rating = 'Unrated'
        }
        dataWithID.id = id
        dataWithID.imdbID = response.data.imdbID
        localStorage[`${id}-rating`] = dataWithID.rating
        localStorage[`${id}-imdb`] = response.data.imdbID
        dispatch(addOMDBData(dataWithID))
      })
      .catch(error => {
        // I know console error's are frowned upon, but I think this one makes sense here.
        console.error('axios error', error)
      })
  }
}
