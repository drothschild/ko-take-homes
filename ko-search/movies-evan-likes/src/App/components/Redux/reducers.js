import { combineReducers } from 'redux'
import { SET_SEARCH_FILTER, SET_DECADE_FILTER, ADD_OMDB_DATA, EXPAND_LINE } from './actions'

const searchFilter = (state = '', action) => {
  if (action.type === SET_SEARCH_FILTER) {
    return action.payload
  }
  return state
}

const decadeFilter = (state = '', action) => {
  if (action.type === SET_DECADE_FILTER) {
    return action.payload
  }
  return state
}

const omdbData = (state = {}, action) => {
  if (action.type === ADD_OMDB_DATA) {
    return Object.assign({}, state, { [action.payload.id]: action.payload })
  }
  return state
}

const lineExpandedID = (state = '', action) => {
  if (action.type === EXPAND_LINE) {
    return action.payload
  }
  return state
}
const rootReducer = combineReducers({ searchFilter, decadeFilter, omdbData, lineExpandedID })

export default rootReducer
