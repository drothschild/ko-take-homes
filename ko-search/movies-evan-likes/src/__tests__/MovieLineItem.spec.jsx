import React from 'react'
import { Provider } from 'react-redux'
import { shallow, render } from 'enzyme'
import MovieLineItem, {
  Unwrapped as UnwrappedMovieLineItem
} from '../App/components/Results/MovieLineItem'
import { expandLine } from '../App/components/Redux/actionCreators'
import store from '../App/components/Redux/store'

const movie = {
  title: 'Arrival',
  year: '2016',
  evanSays:
    "Loved this movie! Cerebral, engaging science fiction at its best. Do yourself a favor and don't watch any trailers or read anything about it before seeing it. You won't be disappointed!"
}
const id = encodeURIComponent(movie.title.toLowerCase() + movie.year)

test('MovieLineItem renders correctly', () => {
  const component = shallow(<UnwrappedMovieLineItem movie={movie} id={id} />)
  expect(component).toMatchSnapshot()
})

test('MovieLineItem should not show "evanSays" initially', () => {
  const component = render(
    <Provider store={store}>
      <MovieLineItem movie={movie} id={id} />
    </Provider>
  )
  const evanCount = 0
  expect(component.find('div').length).toEqual(evanCount)
})

test('MovieLineItem should show "evanSays" after expansion', () => {
  store.dispatch(expandLine(id))
  const component = render(
    <Provider store={store}>
      <MovieLineItem movie={movie} id={id} />
    </Provider>
  )
  store.dispatch(expandLine(''))
  const evanCount = 1
  expect(component.find('div').length).toEqual(evanCount)
})
