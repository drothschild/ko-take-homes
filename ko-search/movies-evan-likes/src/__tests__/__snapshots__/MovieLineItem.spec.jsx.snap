// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieLineItem renders correctly 1`] = `
<li
  onClick={[Function]}
>
  Arrival
   (
  2016
  ) 
  Unrated
</li>
`;
